### DISCUSSION TITLE
Add new file encoding

### DISCUSSION THREAD

**Discussion by BMBurstein** [microsoft/vscode-discussions#132] [Extension Development QnA] [answered]
Is there a way to write an extension that adds a new file encoding? I have a bunch of data files for a very old program that are encoded in DOS-862, which is not available in the encodings list. I didn't find mention of such an option, but I have never written an extension so maybe I am using the docs wrong

**Comment by alefragnani** [comment, marked as answer]
Hi @BMBurstein ,

I guess you should look for [Custom Editor API](https://code.visualstudio.com/api/extension-guides/custom-editors#customeditor-vs-customtexteditor), as you would be responsible for providing the write/read feature for your custom encoded files. Take a look at their sample in https://github.com/microsoft/vscode-extension-samples/tree/main/custom-editor-sample for more details.

On the other hand, I found [an old issue](https://github.com/microsoft/vscode/issues/824), which seems to ask exactly what you need. But it still _open_, with no ETA.

Hope this helps

**Reply by BMBurstein** [reply]
Thank you. I am not looking to add a new file type. I want to add a new file encoding to the list that comes built-in with vscode

**Reply by alefragnani** [reply]
You first asked for an API to write your an extension to support DOS-862 encoding. In this case, you will have to wait for [that old issue](https://github.com/microsoft/vscode/issues/824). The Custom Editor API idea was an alternative solution, in the case you can't wait for that issue to be completed.

On the other hand, your latest comment describes about _built-in encoding list_. In this case, it wouldn't be an extension, but a new feature in VS Code itself. If that's your idea, you must open a new issue in VS Code repo and ask the team to add built-in support for DOS-862 encoding. Maybe they add it by themselves, maybe they ask for contribution (and you could create a PR) and maybe they reject it. 

Hope this helps
